File: lib_views/src/main/java/com/fxc/pics/views/recyclerView/BaseAdapter.java
package com.fxc.pics.views.recyclerView;

import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

/**
 * @author fxc
 * @date 2018/2/24
 */

class BaseAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
	protected RecyclerView.Adapter mAdapter;

	public BaseAdapter(RecyclerView.Adapter adapter) {
		this.mAdapter = adapter;
	}

	@Override
	public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
		return mAdapter.onCreateViewHolder(parent, viewType);
	}

	@SuppressWarnings("unchecked")
	@Override
	public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
		mAdapter.onBindViewHolder(holder, position);
	}

	@Override
	public int getItemCount() {
		return mAdapter.getItemCount();
	}
}

File: lib_views/src/main/java/com/fxc/pics/views/loadMore/BaseAdapter.kt
package com.fxc.adapter

import android.support.v7.widget.RecyclerView

/**
 *
 * @author fxc
 * @date 2018/1/15
 */
abstract class BaseAdapter<T> : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
	private val mData = ArrayList<T>()
	override fun getItemCount(): Int {
		return mData.size
	}

	fun getData():List<T> {
		return mData
	}

	fun refreshData(data: List<T>) {
		if (data.isEmpty()) return
		mData.clear()
		appendData(data)
	}

	fun appendData(data: List<T>) {
		if (data.isEmpty()) return
		mData.addAll(data)
		notifyDataSetChanged()
	}
}